docs/architecture/views.md

```diff
@@ -0,0 +1,169 @@
+# Views
+
+A **view** is a declarative description of what should get rendered and is usually influenced by the current [state](state.md).
+
+It is implemented as a pure function that accepts the current state and returns a [virtual DOM node (VNode)](#virtual-dom). When [state transitions](state.md#state-transitions) happen your views are automatically updated accordingly.
+
+---
+
+## Describing Views
+
+The [`h()`](../api/h.md), [`text()`](../api/text.md), and [`memo()`](../api/memo.md) functions are the building blocks of your views.
+
+[`h()`](../api/h.md) not only describes what HTML elements are being used but also what [actions](actions.md) are wired up if any.
+
+```js
+const view = (state) =>
+  h(
+    "button",
+    {
+      class: { "calling-acid-burn": state.beingFramed },
+      onclick: FindThatDisk,
+    },
+    text("It's in that place where I put that thing that time.")
+  )
+```
+<!-- In the 1995 movie "Hackers", the hacker "The Phantom Freak" calls his friend "Acid Burn" from jail as he's being framed for a crime he didn't commit. -->
+
+[`text()`](../api/text.md) just creates text nodes so the views it can create on its own are necessarily simplistic.
+
+```js
+const view = () => text("Go home and be a family man!")
+```
+<!-- In the videogame "Street Fighter II: The World Warrior", the fighter known as Guile says this taunt to his opponent after defeating them. -->
+
+[`memo()`](../api/memo.md) is designed to be used with other functions that produce VNodes, so it doesn't really describe a view by itself.
+
+```js
+const view = (state) => memo(scenicView, state.vacationSpot)
+```
+<!-- Just a play-on-words between how "view" is used in Hyperapp and everyday language. -->
+
+---
+
+## Components
+
+Views are naturally composable so they can be as simple or complicated as you need. Simpler apps probably just need a single view but in more complicated apps there could be plenty of subviews.
+
+A **component** in Hyperapp can either be a subview or some other function that generates VNodes.
+
+You would typically make components for widgets that provide the building block elements of your app's UI. Components for larger UI segments such as dashboards or pages would make use of these widgets.
+
+In the following example, `coinsDisplay` is a component in the form of a subview while `questionBlock` is a component in the form of some function that returns a VNode. Notice the former cares directly about the state while the latter doesn't:
```

Comment:
I feel like the distinction here between "components", "subviews" and "other functions which return VNodes" is all unnecessary abstraction. All it really gives some definitions/terminology. Consistent terminology is of course good to have when discussiong things with other hyperappers, but I would reserve that for a smaller section to itself labelled "terminology".

In my opinion all the things we need to explain is that there is the "app view" (the function you assign in the app-call) which takes a state and should return a single vnode. This vnode could in turn invoke other functions, which in turn invoke other functions et c. All those functions invoked from the app view, are "view components" because they represent parts that are composed into  a whole, and that whole is the view of some (or multiple) app(s).

Reply:
Yeah, I'll definitely rethink the terminology.  Much of it was influenced by my type definitions in case you haven't noticed. But that doesn't mean it fits perfectly in a documentation context.